### ISSUE TITLE
Support for absolute import paths

### ISSUE THREAD

**Issue by RRamirez1414** [Dschungelabenteuer/vite-plugin-entry-shaking#8]
Hello! 
Our team recently ran into similar degraded performance issues other vite users have mentioned which led us to experiment with this plugin. We are using React + Typescript + Vite.

We configured our tsconfig and viteconfig to resolve and build paths so we can import modules from a directory's shared file. This allows us to import named modules through absolute paths by referencing a directory living directly under the baseUrl path configured in typescript so as long as that module is exported through that directory's shared file.
```javascript
// src/components/index.ts
import a from './a'
import b from './b'
import c from './c'

export { a, b, c }

// src/directory/module.tsx
import { a, b } from 'components'
```
We noticed this plugin doesn't transform named imports from a shared file given an absolute path which leads to runtime errors because the module no longer exists in the shared file. Is this a misuse or a limit of the plugin on our end?

our tsconfig.json
```JSON
{
  "compilerOptions": {
    "baseUrl": "./src",
    ...
```
our vite.config.ts uses [vite-tsconfig-paths](https://github.com/aleclarson/vite-tsconfig-paths) to resolve path mapping
```javascript
export default defineConfig( async () => ({
  plugins: [
    tsconfigPaths(),
    await EntryShakingPlugin({
      targets: [
        resolve(__dirname, 'src/elements'),
        resolve(__dirname, 'src/utils'),
        resolve(__dirname, 'src/hooks'),
      ],
      debug: true,
    }),
  ],
}))
```

**Comment by RRamirez1414** [comment]
Thank you for looking into this! Will do some experimenting on my end, most of our unresolved paths were solved by manually creating resolve aliases through the vite config. I will update this issue if we come across anything else.

**Comment by Idonoghue** [comment]
Hey @Dschungelabenteuer! I'm on the same team as @RRamirez1414. We had our focus shifted elsewhere for a couple months. To answer the absolute imports question, we're indirectly using them because of Typescript's baseURL to construct our full import path. We're not using relative import paths (starting with './' or '../') to reference our targets.

What we missed, and your example highlights, is that we need to add each target path as an alias in tsconfig to resolve it to the full import path:
```
  resolve: {
    alias: {
      components: pathToComponents,
      elements: pathToElements,
      ...etc
    },
  },
```
Once we did that it worked!

Thanks a million for taking the time to publish this. On an average page we've gone from 830 to about 260 JS modules when loading the dev server. ❤️

**Comment by Dschungelabenteuer** [comment]
@Idonoghue Oops I'm sorry for the delay, but I'm really glad it could help, and thank you so much for your feedback and the kind words! It really means a lot ❤️ Please let me know if you run into any further issue/limitation/idea using that plugin, I'm still willing to go 1.x some day  :)

**Comment by Idonoghue** [comment]
@Dschungelabenteuer Now I must apologize to you for the delay haha. I love the simple interface & 'just works' nature of it. I will say the only idea I might have is possibly a feature to automatically opt all index.ts files in as entry points instead of needing to manually define the array. if a hypothetical `targetAll: true` was set you could instead pass a list of index.ts's to ignore?
 
It would basically allow a built-in version of [our usage described here](https://github.com/vitejs/vite/issues/8237#issuecomment-1711103401).

**Comment by Dschungelabenteuer** [comment]
Hey! Better late than never, I've released `0.4.0` which allows you to specify targets using glob patterns, I think this could address your need! I'm closing this issue for now, but feel free to re-open :)
